app/helpers/fhir_helper.rb

```diff
@@ -222,6 +244,12 @@ def patient_from_fhir(patient, default_jurisdiction_id)
       monitoring_plan: from_string_extension(patient, 'Patient', 'monitoring-plan'),
       assigned_user: from_positive_integer_extension(patient, 'Patient', 'assigned-user'),
       additional_planned_travel_start_date: from_date_extension(patient, 'Patient', ['additional-planned-travel-start-date']),
+      additional_planned_travel_end_date: from_date_extension(patient, 'Patient', ['additional-planned-travel-end-date']),
+      additional_planned_travel_destination: from_string_extension(patient, 'Patient', 'additional-planned-travel-destination'),
```

Comment:
Ah this exposed an existing bug. Our implementation of many of the `from_*_extension` needs to be tweaked. When checking for the presence of the extension we do:
```ruby
e.url.include?(extension_id)
```
Which implicitly and wrongly assumes our extension IDs cannot be substrings of other extension IDs. Those should all be changed to something like
```ruby
e.url == SA_EXT_BASE_URL + extension_id
```

Reply:
And to be clear, the bug manifested in this case as: If you set the `additional-planned-travel-destination-country` to, say, "Albania", but you omit the `additional-planned-travel-destination` (ID substring), it will end up setting both the `additional_planned_travel_destination_country` and the `additional_planned_travel_destination` to "Albania".

Reply:
Yikes good catch. Should be fixed in https://github.com/SaraAlert/SaraAlert/pull/1046/commits/ac815a858ba8b9630dccc4f0d3603cc434ac5a64. I had to slightly modify your suggestion to do:
```ruby
e.url == SA_EXT_BASE_URL + extension_id || e.url == extension_id
```
to account for the case that the extension is an inline defined extension, so it has no unique defining URL.